mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java

```diff
@@ -88,8 +78,14 @@ public Builder size(long size) {
         }
 
         public Attachment build() {
-            Preconditions.checkState(attachmentId != null, "'attachmentId' is mandatory");
             Preconditions.checkState(bytes != null, "'bytes' is mandatory");
+            if (attachmentId == null) {
+                attachmentId = AttachmentId.forPayload(bytes);
```

Comment:
calling `build()` should not change anything to the builder state, you have to use local vars for that (or even better, submethods)